### DISCUSSION TITLE
Help decoding Rosstech/DCU706

### DISCUSSION THREAD

**Discussion by suaveolent** [merbanan/rtl_433#2610] [Q&A]
I'm trying to decode the data sent by my spa. See also:
https://github.com/merbanan/rtl_433_tests/pull/442

I already implemtented a decoder for it (https://github.com/suaveolent/rtl_433/commit/a93264412d9dd29a076a0735dec924836eb73b0c)

Given the description in the FCC document (https://fcc.report/FCC-ID/U33-DCU706B/804720.pdf) it seems that each byte is encoded with 11 bits (1 starting bit, one 0 end bit and a parity bit).

I captured two different packages for different temperatures:

Spa Temp Around 34°C
https://triq.org/pdv/#AAB011070100D001A0027004100340082028A48555+AAB020070100D001A0027004100340082028A490A08080B1919291809080C19180808655

Spa Temp Around 18°C
https://triq.org/pdv/#AAB011070100D4019C027403F80324082827B48555+AAB020070100D4019C027403F80324082827B490A08080B1919291809080C19180808655

However, the analyzer shows the same data for both transmissions.

Am I decoding it the wrong way? e.g. wrong short_width / long_width or is the spa simply not transmitting the right data?

Thanks for your help!

**Comment by zuckschwerdt** [comment]
Your decoder timings look plausible. But reset limit here should be 2000 as the maximum run of 1's or 0's is 9.

Try to capture more codes, best case a slow ramp up or down. If you suspect a bug in your decoder use a simple flex decoder like:
` rtl_433 -R 0 -X 'n=name,m=OOK_PCM,s=200,l=200,r=2000,bits>=40'`

**Comment by suaveolent** [comment]
Thanks for the pointer. I captured some new samples today and something is changing now at least. I will try to dig deeper.

**Comment by suaveolent** [comment]
I think I found the culprit: My unit has no temperature probe installed. Hence it always sends the same (wrong) value. So, in theory, my decoder should be correct.
